x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go

```diff
@@ -83,7 +83,7 @@ func (s oneForZeroStrategy) ComputeSwapStep(sqrtPriceCurrent, sqrtPriceNextTick,
 	} else {
 		// amountZero is amountIn.
 		// TODO: multiplication with rounding up at precision end.
-		feeChargeTotal = amountZero.Mul(s.swapFee)
+		feeChargeTotal = amountZero.Mul(s.swapFee).Quo(sdk.OneDec().Sub(s.swapFee))
```

Comment:
what's the reason behind having to divide it by `1-s.swapFee` not `swapFee` itself?

Reply:
We want to charge the fee on the amount in after fee.

By doing `amountZero / (1 - fee)` we get the amount in after fee. By multiplying by swap fee then, we get the desired amount.

For example, assume `amountZero` is 10, `swapFee` is 0.1

If we did `10 * 0.1`, we would get 0.1

However, by doing `10 * / 0.9 * 0.1`, we get `0.1111` which is the fee charged on the full amount needed